tests/proxy_test.go

```diff
@@ -1004,18 +982,22 @@ func (a *unresponsiveAgent) Close() {
 
 // waitForConnectedServerCount waits for the agent ClientSet to have the expected number of health
 // server connections (HealthyClientsCount).
-func waitForConnectedServerCount(t testing.TB, expectedServerCount int, clientset *agent.ClientSet) {
+func waitForConnectedServerCount(t testing.TB, expectedServerCount int, a framework.Agent) {
 	t.Helper()
 	err := wait.PollImmediate(100*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
-		hc := clientset.HealthyClientsCount()
-		if hc == expectedServerCount {
+		csc, err := a.GetConnectedServerCount()
+		if err != nil {
+			return false, err
+		}
+		if csc == expectedServerCount {
 			return true, nil
 		}
 		return false, nil
 	})
 	if err != nil {
-		hc, cc := clientset.HealthyClientsCount(), clientset.ClientsCount()
-		t.Logf("got %d clients, %d of them are healthy; expected %d", cc, hc, expectedServerCount)
+		if csc, err := a.GetConnectedServerCount(); err == nil {
+			t.Logf("got %d server connections; expected %d", csc, expectedServerCount)
+		}
```

Comment:
Note to reviewers: I dropped `ClientsCount` here since it is not exposed by prometheus metrics. It wasn't actually being tested anyway, just used in the error message.